pkg/lib/config/testdata/parse_secret_tests.yaml

```diff
@@ -188,4 +188,96 @@ config:
   secrets:
     - key: bot_protection.provider
       data:
-        type: hcaptcha # not supported
\ No newline at end of file
+        type: hcaptcha # not supported
+
+---
+name: ldap/valid
+error: null
+config:
+  secrets:
+    - key: ldap
+      data:
+        items:
+          - name: test
+            dn: dc=test
+            password: secret
+---
+name: ldap/valid-only-name
+error: null
+config:
+  secrets:
+    - key: ldap
+      data:
+        items:
+          - name: test
+            dn: dc=test
+            password: secret
+---
+name: ldap/missing-required-secret-key
+error: |-
+  invalid secrets:
+  /secrets/0/data/items/0: required
+    map[actual:[dn password] expected:[name] missing:[name]]
+config:
+  secrets:
+    - key: ldap
+      data:
+        items:
+          - dn: dc=test
+            password: secret
+---
+name: ldap/missing-dn-when-password-present
+error: |-
+  invalid secrets:
+  /secrets/0/data/items/0: dependentRequired
```

Comment:
When we use dependentRequired, the error shape changed a bit. This is unexpected. Let's use allOf and if then to implement the condition.


```
{
  "allOf": [
    {
      "if": {
         "required": ["dn"]
       },
       "then": {
          "required": ["password"]
       },
    },
    {
      "if": {
         "required": ["password"]
       },
       "then": {
          "required": ["dn"]
       },
    },
  ]
}

```